File: frontend/event-ticketing-app/src/api/paymentService.ts
import apiClient from './axios';

export interface CheckoutRequest {
  eventId: number;
  items: {
    ticketTypeId: number;
    quantity: number;
  }[];
}

export interface CheckoutResponse {
  sessionId: string;
  sessionUrl: string;
  orderId: number;
}

const paymentService = {
  // Create Stripe checkout session
  createCheckoutSession: async (request: CheckoutRequest): Promise<CheckoutResponse> => {
    const response = await apiClient.post('/payments/create-checkout-session', request);
    return response.data.data;
  },

  // Verify payment session (optional)
  verifySession: async (sessionId: string): Promise<void> => {
    await apiClient.get(`/payments/verify-session/${sessionId}`);
  },
};

export default paymentService;

File: backend/eventticketingplatform/src/main/java/com/nipapager/eventticketingplatform/payment/service/PaymentService.java
package com.nipapager.eventticketingplatform.payment.service;

import com.nipapager.eventticketingplatform.payment.dto.CheckoutResponse;
import com.nipapager.eventticketingplatform.payment.dto.CreateCheckoutRequest;
import com.nipapager.eventticketingplatform.response.Response;

public interface PaymentService {
    Response<CheckoutResponse> createCheckoutSession(CreateCheckoutRequest request);
    void handleWebhook(String payload, String sigHeader);
}